enhancements/sig-architecture/63-hosted-addon/README.md

```diff
@@ -0,0 +1,262 @@
+After transforming, the addon should be deployed successfully either in Default mode or Hosted mode.
+
+1. Consider carefully whether the addon can run in Hosted mode. Will addon functions be affected when addon agent is 
+running outside the managed cluster? If the answer is no, stop.
+2. Refactor the code of the addon agent to accept a flag like `managed-kubeconfig`, and use the kubeconfig received 
+with the flag to build the managed cluster kube client, and use this client to process resources on the managed cluster.
+3. Refactor the deployment manifest of the addon agent, use `if...end...` statement to add the `managed-kubeconfig` 
+flag only in the Hosted mode deploying.
+```
+      ...
+      volumes:
+      {{- if eq .InstallMode "Hosted" }}
+      - name: managed-kubeconfig-secret
+        secret:
+          secretName: helloworld-managed-kubeconfig
+      {{- end }}
+      ...
+      containers:
+      - name: helloworld-agent
+        args:
+          ...
+          {{- if eq .InstallMode "Hosted" }}
+          - "--managed-kubeconfig=/managed/config/kubeconfig"
+          {{- end }}
+          ...
+        volumeMounts:
+          {{- if eq .InstallMode "Hosted" }}
+          - name: managed-kubeconfig-secret
+            mountPath: "/managed/config"
+            readOnly: true
+          {{- end }}
+          ...
+```
+4. Consider each agent manifest need to be deployed where in Hosted mode, and label them with 
+`addon.open-cluster-management.io/manifest-type`, Usually, deployments need to be deployed in the hosting 
+cluster and CRDs owned by the addon need to be deployed in the managed cluster.
+5. Service account, (cluster)role, (cluster)rolebinding are a little special. In general, there is only one copy of 
+these resources in default mode, but now we need to consider splitting them into two copies, one for the hosting 
+cluster (permissions for running the controller, such as operating configmaps, leases, and events), and another for the 
+managed cluster (permissions for the agent to operate the managed cluster resources)
+6. Test the addon deploying and deleting in Hosted mode AND Default mode, make sure it can work on both mode. Default 
+mode can be seen as a special Hosted mode when the managed cluster itself is treated as the hosting cluster.
+
+### Test Plan
+
+E2E tests can use kind to create two clusters to test, one cluster acts as hub and hosting cluster(self-join), and the 
+other acts as the managed cluster, an example should be created using addon framework for the purpose of testing, and 
+the addon agent deployment can be successfully installed on the hosting cluster.
+
+### Graduation Criteria
+
+Alpha:
+* Hosted mode addon is implemented and works well on KinD cluster for at least one addon
+* E2E tests exists
+
+## Alternatives
+
+1. Change the ManagedClusterAddon API to add a field `DeployOption` to describe the deploy option(mode) of the addon.
```

Comment:
I think the field would go well in the spec, it seems similar to the `installNamespace` that is already there.